var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -240,6 +243,64 @@ def check_install(self):
             example = Executable('./conduit_example')
             example()
 
+    def test(self):
+        """
+        Checks the spack install of conduit using conduit's
+        using-with-cmake and using-with-make examples
+        (supports: spack test run)
+        """
+        # can't use self.run_test with the check_install logic
+        # so looks pretty similar 
+        print("Checking Conduit installation...")
+        spec = self.spec
+        install_prefix = self.install_test_root
+        example_src_dir = join_path(install_prefix,
+                                    "examples",
+                                    "conduit",
+                                    "using-with-cmake")
+        print("Checking using-with-cmake example...")
+        with working_dir("check-post-conduit-using-with-cmake-example",
+                         create=True):
+            opts = ["-DCONDUIT_DIR={0}".format(install_prefix),
+                     example_src_dir]
+            opts += self.std_cmake_args
+            opts += self.cmake_args()
+            cmake_args = ["-DCONDUIT_DIR={0}".format(install_prefix),
+                          example_src_dir]
+            # cmake(*cmake_args)
+            # make()
+            cmake_exe = join_path(self.spec['cmake'].prefix,"bin","cmake")
+            self.run_test("/bin/echo",["hello"])
+            self.run_test(cmake_exe,opts)
+            self.run_test(make)
+            self.run_test("conduit_example")
```

Comment:
I *highly* recommend using the `purpose` argument so each execution of `self.run_test` outputs the message before it performs the test.  It can help with debugging, certainly with `graphing the output file.  (FWIW I prefer to prepend `test: ` to each purpose description to make `grep`ing even easier. :grin:)

Reply:
thanks for the suggestion, I added `purpose`